### ISSUE TITLE
backend: Subtitle request crashing server

### ISSUE THREAD

**Issue by ehalmkro** [DianaMukaliyeva/hypertube#57]
Watching a movie (use imdb code `tt0108052`, _Schindler's List_ to reproduce) and clicking the player component's dialog to choose the subtitles sends out the following request:
`http://localhost:5003/api/movies/tt0108052/subtitles/fi/{token}`

This request causes the server to throw an error with the following output:
```
backend_hyper     | (Use `node --trace-deprecation ...` to show where the warning was created)
backend_hyper     | node:_http_outgoing:711
backend_hyper     |     throw new ERR_INVALID_ARG_TYPE(
backend_hyper     |     ^
backend_hyper     | 
backend_hyper     | TypeError [ERR_INVALID_ARG_TYPE]: The "chunk" argument must be of type string or an instance of Buffer or Uint8Array. Received an instance of Error
backend_hyper     |     at new NodeError (node:internal/errors:278:15)
backend_hyper     |     at write_ (node:_http_outgoing:711:11)
backend_hyper     |     at ServerResponse.end (node:_http_outgoing:823:5)
backend_hyper     |     at ReadStream.<anonymous> (file:///app/controllers/subtitlesController.js:18:9)
backend_hyper     |     at ReadStream.emit (node:events:376:20)
backend_hyper     |     at emitErrorNT (node:internal/streams/destroy:188:8)
backend_hyper     |     at emitErrorCloseNT (node:internal/streams/destroy:153:3)
backend_hyper     |     at processTicksAndRejections (node:internal/process/task_queues:80:21) {
backend_hyper     |   code: 'ERR_INVALID_ARG_TYPE'
backend_hyper     | }
```

Assigning this issue to myself for now, will take a look tomorrow. Might have to do with the Finnish locale?

**Comment by ehalmkro** [comment]
Ping @T7Q, just FYI. I'll try to take a look myself first

**Comment by T7Q** [comment]
In which branch this error happened? 
I tried to replicate this error, the master branch didn't throw it. There were no subtitles available for this movie, the frontend does not even send a request to that route to get them.

**Comment by ehalmkro** [comment]
Happened on the backend/movie_stream branch, forgot to specify, sry!

**Comment by T7Q** [comment]
so far subtitles didn't crash ... but got this error ... i'll keep testing after dinner :) 

```
MIDDLEWARE > ERROR HANDLER Error: ENOENT: no such file or directory, stat './movies/tt0068646/The Godfather (1972)/The.Godfather.1972.720p.BrRip.x264.BOKUTOX.YIFY.mp4'
    at Object.statSync (node:fs:1121:3)
    at Object.startFileStream (file:///app/utilities/movieTorrentUtilities.js:20:23)
    at playMovie (file:///app/controllers/movieController.js:81:27)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5) {
  errno: -2,
  syscall: 'stat',
  code: 'ENOENT',
  path: './movies/tt0068646/The Godfather (1972)/The.Godfather.1972.720p.BrRip.x264.BOKUTOX.YIFY.mp4'
}
some error
```

**Comment by ehalmkro** [comment]
That error has to do with the torrent engine not initializing during the 10
second sleep, will hopefully be fixed with the promise rework

**Comment by ehalmkro** [comment]
fixed with 6db56883a339b84a46dcaf8b3ad34f9e1514c677 :D there was one dot in the wrong place
